core/src/test/java/nu/yona/server/crypto/seckey/CryptoSessionTest.java

```diff
@@ -83,46 +80,49 @@ public void decrypt_validPassword_returnsDecryptedData(String plaintext)
 		assertThat(result, equalTo(plaintext));
 	}
 
-	@Test(expected = CryptoException.class)
 	public void decrypt_invalidCryptoVariantNumber_throws()
 	{
 		byte[] initializationVector = new byte[INITIALIZATION_VECTOR_LENGTH];
 		byte[] ciphertext = Base64.getDecoder().decode(encrypt(PASSWORD1, PLAINTEXT1, initializationVector));
 		ciphertext[0] = 13; // Unsupported crypto variant number
 
-		decrypt(PASSWORD1, Base64.getEncoder().encodeToString(ciphertext), initializationVector);
+		assertThrows(CryptoException.class,
+				() -> decrypt(PASSWORD1, Base64.getEncoder().encodeToString(ciphertext), initializationVector));
 	}
 
-	@Test(expected = CryptoException.class)
 	public void decrypt_invalidPassword_throws()
 	{
 		byte[] initializationVector = new byte[INITIALIZATION_VECTOR_LENGTH];
 		String ciphertext = encrypt(PASSWORD1, PLAINTEXT1, initializationVector);
 
-		String plaintext = decrypt(PASSWORD2, ciphertext, initializationVector);
+		assertThrows(CryptoException.class, () -> {
+			String plaintext = decrypt(PASSWORD2, ciphertext, initializationVector);
 
-		// In rare cases, decryption with a wrong password doesn't throw but delivers rubbish.
-		// In such rare cases, compare the string and explicitly throw that exception.
-		assertThat(plaintext, not(equalTo(PLAINTEXT1)));
-		throw CryptoException.decryptingData();
+			// In rare cases, decryption with a wrong password doesn't throw but delivers rubbish.
+			// In such rare cases, compare the string and explicitly throw that exception.
+			assertThat(plaintext, not(equalTo(PLAINTEXT1)));
+			throw CryptoException.decryptingData();
+		});
 	}
 
 	@Test
 	public void start_passwordCheckerSaysOk_doesNotThrow()
 	{
 		try (CryptoSession cryptoSession = CryptoSession.start(Optional.of(PASSWORD1), CryptoSessionTest::passwordIsOk))
 		{
-			assertTrue(true);
+			assertThat("Password checker returned OK", true);
 		}
 	}
 
-	@Test(expected = CryptoException.class)
+	@Test
 	public void start_passwordCheckerSaysNotOk_throws()
 	{
-		try (CryptoSession cryptoSession = CryptoSession.start(Optional.of(PASSWORD1), CryptoSessionTest::passwordIsNotOk))
-		{
-			assertTrue(false);
-		}
+		assertThrows(CryptoException.class, () -> {
+			try (CryptoSession cryptoSession = CryptoSession.start(Optional.of(PASSWORD1), CryptoSessionTest::passwordIsNotOk))
+			{
+				assertThat("Password checker inadvertently returned OK", false);
```

Comment:
We should not use `assertThat` here, but `fail("Password checker inadvertently returned OK")` or just leave it away

Reply:
The assertion is important, so we shouldn't omit it. However, ``fail`` is much better than ``assertThat``.